src/main/scala/com/twitter/finatra/http/SwaggerRouteDSL.scala

```diff
@@ -11,57 +11,68 @@ object SwaggerRouteDSL {
   implicit def convertToSwaggerRouteDSL(dsl: RouteDSL)(implicit swagger: Swagger): SwaggerRouteDSL = new SwaggerRouteDSLWapper(dsl)(swagger)
 }
 
+object PathHelper {
+  implicit class PathWrapper(path: String) {
+    def tailed() = s"/$path/".replaceAll("/{2,}", "/")
+  }
+  implicit class SwaggerWrapper(swagger: Swagger) {
+    def base() = Option(swagger.getBasePath).getOrElse("/")
+  }
+}
+
 trait SwaggerRouteDSL {
+  import PathHelper._
+
   implicit protected val swagger: Swagger
   protected val dsl: RouteDSL
 
   def postWithDoc[RequestType: Manifest, ResponseType: Manifest](route: String, name: String = "", admin: Boolean = false, adminIndexInfo: Option[AdminIndexInfo] = None)
                                                                 (doc: Operation => Unit)
                                                                 (callback: RequestType => ResponseType): Unit = {
-    registerOperation(route, "post")(doc)
-    dsl.post(route, name, admin, adminIndexInfo)(callback)
+    registerOperation(route.tailed, "post")(doc)
```

Comment:
What's the reason for this change? Can you give me an example that the old code does not work.